### PR TITLE
SG provisioning for Ubuntu

### PR DESCRIPTION
**Pull request by raghusarangapani**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:
- SG provisioning for Ubuntu
-

### CODE REVIEW

File: libraries/provision/ansible/playbooks/tasks/fetch-sync-gateway-logs.yml [hunk 1]
```diff
@@ -21,4 +21,16 @@
   become: yes
   # Do not validate checksum as this is just a passive grab of the logs. Since they are continuously being appended to, the checksum may fail
   fetch: src=/home/sync_gateway/logs/sync_gateway_error.log dest=/tmp/sg_logs/{{ inventory_hostname }}/ fail_on_missing=yes flat=yes validate_checksum=no
+  when: pe.stat.exists
+
+- name: SYNC GATEWAY | Check if sync_gateway log is present under /tmp/sg_logs
```
Comment (sridevi-15) [inline libraries/provision/ansible/playbooks/tasks/fetch-sync-gateway-logs.yml]: line 26 to 35, what is this extra logic for?

File: libraries/provision/ansible/playbooks/tasks/remove-sg-accel.yml [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
 - name: SG ACCEL | Stop sg_accel service
   service: name=sg_accel state=stopped
-  when: ansible_distribution == "CentOS" and ansible_distribution_major_version == "7"
+  when: ansible_distribution == "CentOS" and ansible_distribution_major_version == "7" or ansible_distribution == "Ubuntu"
   ignore_errors: yes
 
 - name: SG ACCEL | Stop sg_accel for CentOS 6
   shell: /sbin/initctl stop sg_accel
-  when: (ansible_distribution == "CentOS" and ansible_distribution_major_version == "6")  or ansible_distribution == "RedHat" or ansible_distribution == "Amazon" or ansible_distribution == "Ubuntu"
+  when: (ansible_distribution == "CentOS" and ansible_distribution_major_version == "6")  or ansible_distribution == "RedHat" or ansible_distribution == "Amazon"
```
Comment (sridevi-15) [inline libraries/provision/ansible/playbooks/tasks/remove-sg-accel.yml]: not needed for ubuntu?

File: libraries/provision/ansible/playbooks/tasks/remove-sync-gateway.yml [hunk 3]
```diff
@@ -1,18 +1,25 @@
 # Stop sync_gateway service
 - name: SYNC GATEWAY | Stop sync_gateway service
   service: name=sync_gateway state=stopped
-  when: ansible_distribution == "CentOS" and ansible_distribution_major_version == "7"
+  when: (ansible_distribution == "CentOS" and ansible_distribution_major_version == "7") or ansible_distribution == "Ubuntu"
   ignore_errors: yes
 
 - name: SYNC GATEWAY | Stop sync_gateway for CentOS 6
```
Comment (sridevi-15) [inline libraries/provision/ansible/playbooks/tasks/remove-sync-gateway.yml]: can you replace centos to generic name as codition is added for other platforms? Make change on other files which has conditions for other platforms but name of the task mentioned as centos

File: libraries/provision/ansible/playbooks/tasks/remove-sync-gateway.yml [hunk 4]
```diff
@@ -1,18 +1,25 @@
...
   shell: /sbin/initctl stop sync_gateway
-  when: (ansible_distribution == "CentOS" and ansible_distribution_major_version == "6") or ansible_distribution == "RedHat" or ansible_distribution == "Amazon" or ansible_distribution == "Ubuntu"
+  when: (ansible_distribution == "CentOS" and ansible_distribution_major_version == "6") or ansible_distribution == "RedHat" or ansible_distribution == "Amazon"
```
Comment (sridevi-15) [inline libraries/provision/ansible/playbooks/tasks/remove-sync-gateway.yml]: not needed for ubuntu?

File: libraries/provision/ansible/playbooks/tasks/stop-sg-accel.yml [hunk 5]
```diff
@@ -1,11 +1,11 @@
 # Stop service
 - name: SG ACCEL | stop sg_accel service
   service: name=sg_accel state=stopped
-  when: ansible_distribution == "CentOS" and ansible_distribution_major_version == "7"
+  when: (ansible_distribution == "CentOS" and ansible_distribution_major_version == "7") or (ansible_distribution == "Ubuntu")
 
 - name: SG ACCEL | stop sg_accel for CentOS 6
```
Comment (sridevi-15) [inline libraries/provision/ansible/playbooks/tasks/stop-sg-accel.yml]: Replace Centos to generic name or you can remove it or add other platforms too which added in the condition

File: libraries/provision/ansible/playbooks/tasks/stop-sg-accel.yml [hunk 6]
```diff
@@ -1,11 +1,11 @@
...
   shell: /sbin/initctl stop sg_accel
-  when: (ansible_distribution == "CentOS" and ansible_distribution_major_version == "6") or ansible_distribution == "RedHat" or ansible_distribution == "Amazon" or ansible_distribution == "Ubuntu"
+  when: (ansible_distribution == "CentOS" and ansible_distribution_major_version == "6") or ansible_distribution == "RedHat" or ansible_distribution == "Amazon"
```
Comment (sridevi-15) [inline libraries/provision/ansible/playbooks/tasks/stop-sg-accel.yml]: same as above, not needed for ubuntu

**Comment by hrajput89** [comment]
I'll create a new PR as this branch doesn't have all the changes

**Pull request closed without merging**